ThirdParty/GNUmakefile

```diff
@@ -0,0 +1,95 @@
+#
+# Build configuration
+#
+
+COMP            = gcc
+NVCC            = nvcc
+DEBUG           = FALSE
+USE_CUDA        = FALSE
+USE_KLU         = FALSE
+
+#SS_VERSION=5.7.2
+SS_VERSION=5.4.0
+
+#include $(AMREX_HOME)/Tools/GNUMake/Make.defs
+
+tp_suffix=$(COMP)$(DebugSuffix)$(CUDASuffix)$(KLUSuffix)
+INSTALL_PREFIX=$(PWD)/INSTALL/$(tp_suffix)
+
+ifeq ($(USE_KLU),TRUE)
+  KLUSuffix   := .KLU
+  KLU_DEFINES=-DKLU_ENABLE:BOOL=ON -DKLU_INCLUDE_DIR=${INSTALL_PREFIX}/include -DKLU_LIBRARY_DIR=${INSTALL_PREFIX}/lib 
+  SS_BUILD_PREFIX=$(PWD)/BUILD/SUITESPARSE
+
+  SS_DIST_DIR=$(SS_BUILD_PREFIX)/dist
+  SS_SOURCE_DIR=$(SS_BUILD_PREFIX)/SuiteSparse-$(SS_VERSION)
+  SS_BUILD_DIR=$(SS_BUILD_PREFIX)/build/$(tp_suffix)
+
+  ONE_SS_SRC_FILE=$(SS_SOURCE_DIR)/LICENSE.txt
+  ONE_SS_LIB_FILE=$(INSTALL_PREFIX)/libamd.so
+  SS_DIST_FILE=v$(SS_VERSION).tar.gz
+else
+  KLUSuffix=
+  KLU_DEFINES=-DKLU_ENABLE:BOOL=OFF
+  ONE_SS_SRC_FILE=
+  ONE_SS_LIB_FILE=
+endif
+
+ifeq ($(DEBUG),FALSE)
+  BUILD_TYPE=Release
+  COMP_FLAGS="-O3 -DNDEBUG"
+  DebugSuffix=
+else
+  BUILD_TYPE=Debug
+  DebugSuffix=.DEBUG
+endif
+
+ifeq ($(USE_CUDA),TRUE)
+  CUDA_DEFINES=-DCUDA_ENABLE:BOOL=ON -DCUDA_HOST_COMPILER=$(NVCC)
+  CUDASuffix=.CUDA
+else
+  CUDA_DEFINES=-DCUDA_ENABLE:BOOL=OFF
+  CUDASuffix=
+endif
+
+SUNDIALS_BUILD_PREFIX=$(PWD)/BUILD/SUNDIALS
+SUNDIALS_SOURCE_DIR=$(SUNDIALS_BUILD_PREFIX)/src
+SUNDIALS_BUILD_DIR=$(SUNDIALS_BUILD_PREFIX)/build/$(tp_suffix)
+ONE_SUNDIALS_SRC_FILE=$(SUNDIALS_SOURCE_DIR)/config
+ONE_SUNDIALS_LIB_FILE=$(INSTALL_PREFIX)/lib/libsundials_cvodes.a
+
+all: $(ONE_SUNDIALS_LIB_FILE)
+
+tp_suffix:
+	@echo $(tp_suffix)
+
+
+$(ONE_SUNDIALS_LIB_FILE): $(ONE_SUNDIALS_SRC_FILE) $(ONE_SS_LIB_FILE)
+	@echo Target $(ONE_SUNDIALS_LIB_FILE)
+	@mkdir -p $(SUNDIALS_BUILD_DIR)
+	cd $(SUNDIALS_BUILD_DIR); cmake -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX) -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_C_COMPILER=$(COMP) $(CUDA_HOST_COMP_DEFINE) -DEXAMPLES_INSTALL_PATH=$(INSTALL_PREFIX)/examples -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DCMAKE_C_FLAGS_RELEASE=$(COMP_FLAGS) $(CUDA_DEFINES) -DMPI_ENABLE:BOOL=OFF -DOPENMP_ENABLE:BOOL=OFF $(KLU_DEFINES) -DEXAMPLES_ENABLE_C:BOOL=OFF $(SUNDIALS_SOURCE_DIR); make; make install
```

Comment:
Is there a possibility here to generate a txt file that summarizes the options the user actually chose to build sundials ? I know we will ultimately have the CMakeCache in the builddir but I think it might be good for debug and verification purposes

Reply:
I stand by this one though !

Reply:
Accidentally left debugging in.
Re: question. Anything I try to add will be a hacked version of what is actually used.  The CMake record should be sufficient.